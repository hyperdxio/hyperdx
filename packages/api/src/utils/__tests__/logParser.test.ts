import { mapObjectToKeyValuePairs, traverseJson } from '../logParser';

describe('logParser', () => {
  it('traverseJson', () => {
    const jsonIt = traverseJson({
      foo: {
        bar: 'bar',
        foo1: {
          foo1: 'bar1',
          foo2: {
            bar2: 'bar2',
          },
        },
      },
    });
    const keys: any[] = [];
    const values: any[] = [];
    for (const [key, value] of jsonIt) {
      keys.push(key);
      values.push(value);
    }

    expect(keys).toEqual([
      ['foo'],
      ['foo', 'bar'],
      ['foo', 'foo1'],
      ['foo', 'foo1', 'foo1'],
      ['foo', 'foo1', 'foo2'],
      ['foo', 'foo1', 'foo2', 'bar2'],
    ]);
    expect(values).toEqual([
      {
        bar: 'bar',
        foo1: {
          foo1: 'bar1',
          foo2: {
            bar2: 'bar2',
          },
        },
      },
      'bar',
      {
        foo1: 'bar1',
        foo2: {
          bar2: 'bar2',
        },
      },
      'bar1',
      {
        bar2: 'bar2',
      },
      'bar2',
    ]);
  });

  it('mapObjectToKeyValuePairs', () => {
    expect(mapObjectToKeyValuePairs(null as any)).toEqual({
      'bool.names': [],
      'bool.values': [],
      'number.names': [],
      'number.values': [],
      'string.names': [],
      'string.values': [],
    });

    expect(mapObjectToKeyValuePairs({})).toEqual({
      'bool.names': [],
      'bool.values': [],
      'number.names': [],
      'number.values': [],
      'string.names': [],
      'string.values': [],
    });

    expect(
      mapObjectToKeyValuePairs({ foo: '123', foo1: 123, foo2: false }),
    ).toEqual({
      'bool.names': ['foo2'],
      'bool.values': [0],
      'number.names': ['foo1'],
      'number.values': [123],
      'string.names': ['foo'],
      'string.values': ['123'],
    });

    expect(
      mapObjectToKeyValuePairs({
        foo: '123',
        foo1: 123,
        foo2: false,
        nested: { foo: 'bar' },
        good: {
          burrito: {
            is: true,
          },
        },
        array1: [456],
        array2: [
          'foo1',
          {
            foo2: 'bar2',
          },
          [
            {
              foo3: 'bar3',
            },
          ],
        ],
      }),
    ).toMatchSnapshot();

    const testObject = {};
    for (let i = 0; i < 2000; i++) {
      testObject[`foo${i}`] = i;
    }
    const result = mapObjectToKeyValuePairs(testObject);
    expect(result['number.names'].length).toEqual(1024);
    expect(result['number.values'].length).toEqual(1024);
    expect(result).toMatchSnapshot();
  });
});
