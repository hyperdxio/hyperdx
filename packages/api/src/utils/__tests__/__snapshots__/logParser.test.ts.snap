// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logParser mapObjectToKeyValuePairs 1`] = `
Object {
  "bool.names": Array [
    "foo2",
    "good.burrito.is",
  ],
  "bool.values": Array [
    0,
    1,
  ],
  "number.names": Array [
    "foo1",
  ],
  "number.values": Array [
    123,
  ],
  "string.names": Array [
    "foo",
    "nested.foo",
    "array1",
    "array2",
  ],
  "string.values": Array [
    "123",
    "bar",
    "[456]",
    "[\\"foo1\\",{\\"foo2\\":\\"bar2\\"},[{\\"foo3\\":\\"bar3\\"}]]",
  ],
}
`;

exports[`logParser mapObjectToKeyValuePairs 2`] = `
Object {
  "bool.names": Array [],
  "bool.values": Array [],
  "number.names": Array [
    "foo0",
    "foo1",
    "foo2",
    "foo3",
    "foo4",
    "foo5",
    "foo6",
    "foo7",
    "foo8",
    "foo9",
    "foo10",
    "foo11",
    "foo12",
    "foo13",
    "foo14",
    "foo15",
    "foo16",
    "foo17",
    "foo18",
    "foo19",
    "foo20",
    "foo21",
    "foo22",
    "foo23",
    "foo24",
    "foo25",
    "foo26",
    "foo27",
    "foo28",
    "foo29",
    "foo30",
    "foo31",
    "foo32",
    "foo33",
    "foo34",
    "foo35",
    "foo36",
    "foo37",
    "foo38",
    "foo39",
    "foo40",
    "foo41",
    "foo42",
    "foo43",
    "foo44",
    "foo45",
    "foo46",
    "foo47",
    "foo48",
    "foo49",
    "foo50",
    "foo51",
    "foo52",
    "foo53",
    "foo54",
    "foo55",
    "foo56",
    "foo57",
    "foo58",
    "foo59",
    "foo60",
    "foo61",
    "foo62",
    "foo63",
    "foo64",
    "foo65",
    "foo66",
    "foo67",
    "foo68",
    "foo69",
    "foo70",
    "foo71",
    "foo72",
    "foo73",
    "foo74",
    "foo75",
    "foo76",
    "foo77",
    "foo78",
    "foo79",
    "foo80",
    "foo81",
    "foo82",
    "foo83",
    "foo84",
    "foo85",
    "foo86",
    "foo87",
    "foo88",
    "foo89",
    "foo90",
    "foo91",
    "foo92",
    "foo93",
    "foo94",
    "foo95",
    "foo96",
    "foo97",
    "foo98",
    "foo99",
    "foo100",
    "foo101",
    "foo102",
    "foo103",
    "foo104",
    "foo105",
    "foo106",
    "foo107",
    "foo108",
    "foo109",
    "foo110",
    "foo111",
    "foo112",
    "foo113",
    "foo114",
    "foo115",
    "foo116",
    "foo117",
    "foo118",
    "foo119",
    "foo120",
    "foo121",
    "foo122",
    "foo123",
    "foo124",
    "foo125",
    "foo126",
    "foo127",
    "foo128",
    "foo129",
    "foo130",
    "foo131",
    "foo132",
    "foo133",
    "foo134",
    "foo135",
    "foo136",
    "foo137",
    "foo138",
    "foo139",
    "foo140",
    "foo141",
    "foo142",
    "foo143",
    "foo144",
    "foo145",
    "foo146",
    "foo147",
    "foo148",
    "foo149",
    "foo150",
    "foo151",
    "foo152",
    "foo153",
    "foo154",
    "foo155",
    "foo156",
    "foo157",
    "foo158",
    "foo159",
    "foo160",
    "foo161",
    "foo162",
    "foo163",
    "foo164",
    "foo165",
    "foo166",
    "foo167",
    "foo168",
    "foo169",
    "foo170",
    "foo171",
    "foo172",
    "foo173",
    "foo174",
    "foo175",
    "foo176",
    "foo177",
    "foo178",
    "foo179",
    "foo180",
    "foo181",
    "foo182",
    "foo183",
    "foo184",
    "foo185",
    "foo186",
    "foo187",
    "foo188",
    "foo189",
    "foo190",
    "foo191",
    "foo192",
    "foo193",
    "foo194",
    "foo195",
    "foo196",
    "foo197",
    "foo198",
    "foo199",
    "foo200",
    "foo201",
    "foo202",
    "foo203",
    "foo204",
    "foo205",
    "foo206",
    "foo207",
    "foo208",
    "foo209",
    "foo210",
    "foo211",
    "foo212",
    "foo213",
    "foo214",
    "foo215",
    "foo216",
    "foo217",
    "foo218",
    "foo219",
    "foo220",
    "foo221",
    "foo222",
    "foo223",
    "foo224",
    "foo225",
    "foo226",
    "foo227",
    "foo228",
    "foo229",
    "foo230",
    "foo231",
    "foo232",
    "foo233",
    "foo234",
    "foo235",
    "foo236",
    "foo237",
    "foo238",
    "foo239",
    "foo240",
    "foo241",
    "foo242",
    "foo243",
    "foo244",
    "foo245",
    "foo246",
    "foo247",
    "foo248",
    "foo249",
    "foo250",
    "foo251",
    "foo252",
    "foo253",
    "foo254",
    "foo255",
    "foo256",
    "foo257",
    "foo258",
    "foo259",
    "foo260",
    "foo261",
    "foo262",
    "foo263",
    "foo264",
    "foo265",
    "foo266",
    "foo267",
    "foo268",
    "foo269",
    "foo270",
    "foo271",
    "foo272",
    "foo273",
    "foo274",
    "foo275",
    "foo276",
    "foo277",
    "foo278",
    "foo279",
    "foo280",
    "foo281",
    "foo282",
    "foo283",
    "foo284",
    "foo285",
    "foo286",
    "foo287",
    "foo288",
    "foo289",
    "foo290",
    "foo291",
    "foo292",
    "foo293",
    "foo294",
    "foo295",
    "foo296",
    "foo297",
    "foo298",
    "foo299",
    "foo300",
    "foo301",
    "foo302",
    "foo303",
    "foo304",
    "foo305",
    "foo306",
    "foo307",
    "foo308",
    "foo309",
    "foo310",
    "foo311",
    "foo312",
    "foo313",
    "foo314",
    "foo315",
    "foo316",
    "foo317",
    "foo318",
    "foo319",
    "foo320",
    "foo321",
    "foo322",
    "foo323",
    "foo324",
    "foo325",
    "foo326",
    "foo327",
    "foo328",
    "foo329",
    "foo330",
    "foo331",
    "foo332",
    "foo333",
    "foo334",
    "foo335",
    "foo336",
    "foo337",
    "foo338",
    "foo339",
    "foo340",
    "foo341",
    "foo342",
    "foo343",
    "foo344",
    "foo345",
    "foo346",
    "foo347",
    "foo348",
    "foo349",
    "foo350",
    "foo351",
    "foo352",
    "foo353",
    "foo354",
    "foo355",
    "foo356",
    "foo357",
    "foo358",
    "foo359",
    "foo360",
    "foo361",
    "foo362",
    "foo363",
    "foo364",
    "foo365",
    "foo366",
    "foo367",
    "foo368",
    "foo369",
    "foo370",
    "foo371",
    "foo372",
    "foo373",
    "foo374",
    "foo375",
    "foo376",
    "foo377",
    "foo378",
    "foo379",
    "foo380",
    "foo381",
    "foo382",
    "foo383",
    "foo384",
    "foo385",
    "foo386",
    "foo387",
    "foo388",
    "foo389",
    "foo390",
    "foo391",
    "foo392",
    "foo393",
    "foo394",
    "foo395",
    "foo396",
    "foo397",
    "foo398",
    "foo399",
    "foo400",
    "foo401",
    "foo402",
    "foo403",
    "foo404",
    "foo405",
    "foo406",
    "foo407",
    "foo408",
    "foo409",
    "foo410",
    "foo411",
    "foo412",
    "foo413",
    "foo414",
    "foo415",
    "foo416",
    "foo417",
    "foo418",
    "foo419",
    "foo420",
    "foo421",
    "foo422",
    "foo423",
    "foo424",
    "foo425",
    "foo426",
    "foo427",
    "foo428",
    "foo429",
    "foo430",
    "foo431",
    "foo432",
    "foo433",
    "foo434",
    "foo435",
    "foo436",
    "foo437",
    "foo438",
    "foo439",
    "foo440",
    "foo441",
    "foo442",
    "foo443",
    "foo444",
    "foo445",
    "foo446",
    "foo447",
    "foo448",
    "foo449",
    "foo450",
    "foo451",
    "foo452",
    "foo453",
    "foo454",
    "foo455",
    "foo456",
    "foo457",
    "foo458",
    "foo459",
    "foo460",
    "foo461",
    "foo462",
    "foo463",
    "foo464",
    "foo465",
    "foo466",
    "foo467",
    "foo468",
    "foo469",
    "foo470",
    "foo471",
    "foo472",
    "foo473",
    "foo474",
    "foo475",
    "foo476",
    "foo477",
    "foo478",
    "foo479",
    "foo480",
    "foo481",
    "foo482",
    "foo483",
    "foo484",
    "foo485",
    "foo486",
    "foo487",
    "foo488",
    "foo489",
    "foo490",
    "foo491",
    "foo492",
    "foo493",
    "foo494",
    "foo495",
    "foo496",
    "foo497",
    "foo498",
    "foo499",
    "foo500",
    "foo501",
    "foo502",
    "foo503",
    "foo504",
    "foo505",
    "foo506",
    "foo507",
    "foo508",
    "foo509",
    "foo510",
    "foo511",
    "foo512",
    "foo513",
    "foo514",
    "foo515",
    "foo516",
    "foo517",
    "foo518",
    "foo519",
    "foo520",
    "foo521",
    "foo522",
    "foo523",
    "foo524",
    "foo525",
    "foo526",
    "foo527",
    "foo528",
    "foo529",
    "foo530",
    "foo531",
    "foo532",
    "foo533",
    "foo534",
    "foo535",
    "foo536",
    "foo537",
    "foo538",
    "foo539",
    "foo540",
    "foo541",
    "foo542",
    "foo543",
    "foo544",
    "foo545",
    "foo546",
    "foo547",
    "foo548",
    "foo549",
    "foo550",
    "foo551",
    "foo552",
    "foo553",
    "foo554",
    "foo555",
    "foo556",
    "foo557",
    "foo558",
    "foo559",
    "foo560",
    "foo561",
    "foo562",
    "foo563",
    "foo564",
    "foo565",
    "foo566",
    "foo567",
    "foo568",
    "foo569",
    "foo570",
    "foo571",
    "foo572",
    "foo573",
    "foo574",
    "foo575",
    "foo576",
    "foo577",
    "foo578",
    "foo579",
    "foo580",
    "foo581",
    "foo582",
    "foo583",
    "foo584",
    "foo585",
    "foo586",
    "foo587",
    "foo588",
    "foo589",
    "foo590",
    "foo591",
    "foo592",
    "foo593",
    "foo594",
    "foo595",
    "foo596",
    "foo597",
    "foo598",
    "foo599",
    "foo600",
    "foo601",
    "foo602",
    "foo603",
    "foo604",
    "foo605",
    "foo606",
    "foo607",
    "foo608",
    "foo609",
    "foo610",
    "foo611",
    "foo612",
    "foo613",
    "foo614",
    "foo615",
    "foo616",
    "foo617",
    "foo618",
    "foo619",
    "foo620",
    "foo621",
    "foo622",
    "foo623",
    "foo624",
    "foo625",
    "foo626",
    "foo627",
    "foo628",
    "foo629",
    "foo630",
    "foo631",
    "foo632",
    "foo633",
    "foo634",
    "foo635",
    "foo636",
    "foo637",
    "foo638",
    "foo639",
    "foo640",
    "foo641",
    "foo642",
    "foo643",
    "foo644",
    "foo645",
    "foo646",
    "foo647",
    "foo648",
    "foo649",
    "foo650",
    "foo651",
    "foo652",
    "foo653",
    "foo654",
    "foo655",
    "foo656",
    "foo657",
    "foo658",
    "foo659",
    "foo660",
    "foo661",
    "foo662",
    "foo663",
    "foo664",
    "foo665",
    "foo666",
    "foo667",
    "foo668",
    "foo669",
    "foo670",
    "foo671",
    "foo672",
    "foo673",
    "foo674",
    "foo675",
    "foo676",
    "foo677",
    "foo678",
    "foo679",
    "foo680",
    "foo681",
    "foo682",
    "foo683",
    "foo684",
    "foo685",
    "foo686",
    "foo687",
    "foo688",
    "foo689",
    "foo690",
    "foo691",
    "foo692",
    "foo693",
    "foo694",
    "foo695",
    "foo696",
    "foo697",
    "foo698",
    "foo699",
    "foo700",
    "foo701",
    "foo702",
    "foo703",
    "foo704",
    "foo705",
    "foo706",
    "foo707",
    "foo708",
    "foo709",
    "foo710",
    "foo711",
    "foo712",
    "foo713",
    "foo714",
    "foo715",
    "foo716",
    "foo717",
    "foo718",
    "foo719",
    "foo720",
    "foo721",
    "foo722",
    "foo723",
    "foo724",
    "foo725",
    "foo726",
    "foo727",
    "foo728",
    "foo729",
    "foo730",
    "foo731",
    "foo732",
    "foo733",
    "foo734",
    "foo735",
    "foo736",
    "foo737",
    "foo738",
    "foo739",
    "foo740",
    "foo741",
    "foo742",
    "foo743",
    "foo744",
    "foo745",
    "foo746",
    "foo747",
    "foo748",
    "foo749",
    "foo750",
    "foo751",
    "foo752",
    "foo753",
    "foo754",
    "foo755",
    "foo756",
    "foo757",
    "foo758",
    "foo759",
    "foo760",
    "foo761",
    "foo762",
    "foo763",
    "foo764",
    "foo765",
    "foo766",
    "foo767",
    "foo768",
    "foo769",
    "foo770",
    "foo771",
    "foo772",
    "foo773",
    "foo774",
    "foo775",
    "foo776",
    "foo777",
    "foo778",
    "foo779",
    "foo780",
    "foo781",
    "foo782",
    "foo783",
    "foo784",
    "foo785",
    "foo786",
    "foo787",
    "foo788",
    "foo789",
    "foo790",
    "foo791",
    "foo792",
    "foo793",
    "foo794",
    "foo795",
    "foo796",
    "foo797",
    "foo798",
    "foo799",
    "foo800",
    "foo801",
    "foo802",
    "foo803",
    "foo804",
    "foo805",
    "foo806",
    "foo807",
    "foo808",
    "foo809",
    "foo810",
    "foo811",
    "foo812",
    "foo813",
    "foo814",
    "foo815",
    "foo816",
    "foo817",
    "foo818",
    "foo819",
    "foo820",
    "foo821",
    "foo822",
    "foo823",
    "foo824",
    "foo825",
    "foo826",
    "foo827",
    "foo828",
    "foo829",
    "foo830",
    "foo831",
    "foo832",
    "foo833",
    "foo834",
    "foo835",
    "foo836",
    "foo837",
    "foo838",
    "foo839",
    "foo840",
    "foo841",
    "foo842",
    "foo843",
    "foo844",
    "foo845",
    "foo846",
    "foo847",
    "foo848",
    "foo849",
    "foo850",
    "foo851",
    "foo852",
    "foo853",
    "foo854",
    "foo855",
    "foo856",
    "foo857",
    "foo858",
    "foo859",
    "foo860",
    "foo861",
    "foo862",
    "foo863",
    "foo864",
    "foo865",
    "foo866",
    "foo867",
    "foo868",
    "foo869",
    "foo870",
    "foo871",
    "foo872",
    "foo873",
    "foo874",
    "foo875",
    "foo876",
    "foo877",
    "foo878",
    "foo879",
    "foo880",
    "foo881",
    "foo882",
    "foo883",
    "foo884",
    "foo885",
    "foo886",
    "foo887",
    "foo888",
    "foo889",
    "foo890",
    "foo891",
    "foo892",
    "foo893",
    "foo894",
    "foo895",
    "foo896",
    "foo897",
    "foo898",
    "foo899",
    "foo900",
    "foo901",
    "foo902",
    "foo903",
    "foo904",
    "foo905",
    "foo906",
    "foo907",
    "foo908",
    "foo909",
    "foo910",
    "foo911",
    "foo912",
    "foo913",
    "foo914",
    "foo915",
    "foo916",
    "foo917",
    "foo918",
    "foo919",
    "foo920",
    "foo921",
    "foo922",
    "foo923",
    "foo924",
    "foo925",
    "foo926",
    "foo927",
    "foo928",
    "foo929",
    "foo930",
    "foo931",
    "foo932",
    "foo933",
    "foo934",
    "foo935",
    "foo936",
    "foo937",
    "foo938",
    "foo939",
    "foo940",
    "foo941",
    "foo942",
    "foo943",
    "foo944",
    "foo945",
    "foo946",
    "foo947",
    "foo948",
    "foo949",
    "foo950",
    "foo951",
    "foo952",
    "foo953",
    "foo954",
    "foo955",
    "foo956",
    "foo957",
    "foo958",
    "foo959",
    "foo960",
    "foo961",
    "foo962",
    "foo963",
    "foo964",
    "foo965",
    "foo966",
    "foo967",
    "foo968",
    "foo969",
    "foo970",
    "foo971",
    "foo972",
    "foo973",
    "foo974",
    "foo975",
    "foo976",
    "foo977",
    "foo978",
    "foo979",
    "foo980",
    "foo981",
    "foo982",
    "foo983",
    "foo984",
    "foo985",
    "foo986",
    "foo987",
    "foo988",
    "foo989",
    "foo990",
    "foo991",
    "foo992",
    "foo993",
    "foo994",
    "foo995",
    "foo996",
    "foo997",
    "foo998",
    "foo999",
    "foo1000",
    "foo1001",
    "foo1002",
    "foo1003",
    "foo1004",
    "foo1005",
    "foo1006",
    "foo1007",
    "foo1008",
    "foo1009",
    "foo1010",
    "foo1011",
    "foo1012",
    "foo1013",
    "foo1014",
    "foo1015",
    "foo1016",
    "foo1017",
    "foo1018",
    "foo1019",
    "foo1020",
    "foo1021",
    "foo1022",
    "foo1023",
  ],
  "number.values": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
    24,
    25,
    26,
    27,
    28,
    29,
    30,
    31,
    32,
    33,
    34,
    35,
    36,
    37,
    38,
    39,
    40,
    41,
    42,
    43,
    44,
    45,
    46,
    47,
    48,
    49,
    50,
    51,
    52,
    53,
    54,
    55,
    56,
    57,
    58,
    59,
    60,
    61,
    62,
    63,
    64,
    65,
    66,
    67,
    68,
    69,
    70,
    71,
    72,
    73,
    74,
    75,
    76,
    77,
    78,
    79,
    80,
    81,
    82,
    83,
    84,
    85,
    86,
    87,
    88,
    89,
    90,
    91,
    92,
    93,
    94,
    95,
    96,
    97,
    98,
    99,
    100,
    101,
    102,
    103,
    104,
    105,
    106,
    107,
    108,
    109,
    110,
    111,
    112,
    113,
    114,
    115,
    116,
    117,
    118,
    119,
    120,
    121,
    122,
    123,
    124,
    125,
    126,
    127,
    128,
    129,
    130,
    131,
    132,
    133,
    134,
    135,
    136,
    137,
    138,
    139,
    140,
    141,
    142,
    143,
    144,
    145,
    146,
    147,
    148,
    149,
    150,
    151,
    152,
    153,
    154,
    155,
    156,
    157,
    158,
    159,
    160,
    161,
    162,
    163,
    164,
    165,
    166,
    167,
    168,
    169,
    170,
    171,
    172,
    173,
    174,
    175,
    176,
    177,
    178,
    179,
    180,
    181,
    182,
    183,
    184,
    185,
    186,
    187,
    188,
    189,
    190,
    191,
    192,
    193,
    194,
    195,
    196,
    197,
    198,
    199,
    200,
    201,
    202,
    203,
    204,
    205,
    206,
    207,
    208,
    209,
    210,
    211,
    212,
    213,
    214,
    215,
    216,
    217,
    218,
    219,
    220,
    221,
    222,
    223,
    224,
    225,
    226,
    227,
    228,
    229,
    230,
    231,
    232,
    233,
    234,
    235,
    236,
    237,
    238,
    239,
    240,
    241,
    242,
    243,
    244,
    245,
    246,
    247,
    248,
    249,
    250,
    251,
    252,
    253,
    254,
    255,
    256,
    257,
    258,
    259,
    260,
    261,
    262,
    263,
    264,
    265,
    266,
    267,
    268,
    269,
    270,
    271,
    272,
    273,
    274,
    275,
    276,
    277,
    278,
    279,
    280,
    281,
    282,
    283,
    284,
    285,
    286,
    287,
    288,
    289,
    290,
    291,
    292,
    293,
    294,
    295,
    296,
    297,
    298,
    299,
    300,
    301,
    302,
    303,
    304,
    305,
    306,
    307,
    308,
    309,
    310,
    311,
    312,
    313,
    314,
    315,
    316,
    317,
    318,
    319,
    320,
    321,
    322,
    323,
    324,
    325,
    326,
    327,
    328,
    329,
    330,
    331,
    332,
    333,
    334,
    335,
    336,
    337,
    338,
    339,
    340,
    341,
    342,
    343,
    344,
    345,
    346,
    347,
    348,
    349,
    350,
    351,
    352,
    353,
    354,
    355,
    356,
    357,
    358,
    359,
    360,
    361,
    362,
    363,
    364,
    365,
    366,
    367,
    368,
    369,
    370,
    371,
    372,
    373,
    374,
    375,
    376,
    377,
    378,
    379,
    380,
    381,
    382,
    383,
    384,
    385,
    386,
    387,
    388,
    389,
    390,
    391,
    392,
    393,
    394,
    395,
    396,
    397,
    398,
    399,
    400,
    401,
    402,
    403,
    404,
    405,
    406,
    407,
    408,
    409,
    410,
    411,
    412,
    413,
    414,
    415,
    416,
    417,
    418,
    419,
    420,
    421,
    422,
    423,
    424,
    425,
    426,
    427,
    428,
    429,
    430,
    431,
    432,
    433,
    434,
    435,
    436,
    437,
    438,
    439,
    440,
    441,
    442,
    443,
    444,
    445,
    446,
    447,
    448,
    449,
    450,
    451,
    452,
    453,
    454,
    455,
    456,
    457,
    458,
    459,
    460,
    461,
    462,
    463,
    464,
    465,
    466,
    467,
    468,
    469,
    470,
    471,
    472,
    473,
    474,
    475,
    476,
    477,
    478,
    479,
    480,
    481,
    482,
    483,
    484,
    485,
    486,
    487,
    488,
    489,
    490,
    491,
    492,
    493,
    494,
    495,
    496,
    497,
    498,
    499,
    500,
    501,
    502,
    503,
    504,
    505,
    506,
    507,
    508,
    509,
    510,
    511,
    512,
    513,
    514,
    515,
    516,
    517,
    518,
    519,
    520,
    521,
    522,
    523,
    524,
    525,
    526,
    527,
    528,
    529,
    530,
    531,
    532,
    533,
    534,
    535,
    536,
    537,
    538,
    539,
    540,
    541,
    542,
    543,
    544,
    545,
    546,
    547,
    548,
    549,
    550,
    551,
    552,
    553,
    554,
    555,
    556,
    557,
    558,
    559,
    560,
    561,
    562,
    563,
    564,
    565,
    566,
    567,
    568,
    569,
    570,
    571,
    572,
    573,
    574,
    575,
    576,
    577,
    578,
    579,
    580,
    581,
    582,
    583,
    584,
    585,
    586,
    587,
    588,
    589,
    590,
    591,
    592,
    593,
    594,
    595,
    596,
    597,
    598,
    599,
    600,
    601,
    602,
    603,
    604,
    605,
    606,
    607,
    608,
    609,
    610,
    611,
    612,
    613,
    614,
    615,
    616,
    617,
    618,
    619,
    620,
    621,
    622,
    623,
    624,
    625,
    626,
    627,
    628,
    629,
    630,
    631,
    632,
    633,
    634,
    635,
    636,
    637,
    638,
    639,
    640,
    641,
    642,
    643,
    644,
    645,
    646,
    647,
    648,
    649,
    650,
    651,
    652,
    653,
    654,
    655,
    656,
    657,
    658,
    659,
    660,
    661,
    662,
    663,
    664,
    665,
    666,
    667,
    668,
    669,
    670,
    671,
    672,
    673,
    674,
    675,
    676,
    677,
    678,
    679,
    680,
    681,
    682,
    683,
    684,
    685,
    686,
    687,
    688,
    689,
    690,
    691,
    692,
    693,
    694,
    695,
    696,
    697,
    698,
    699,
    700,
    701,
    702,
    703,
    704,
    705,
    706,
    707,
    708,
    709,
    710,
    711,
    712,
    713,
    714,
    715,
    716,
    717,
    718,
    719,
    720,
    721,
    722,
    723,
    724,
    725,
    726,
    727,
    728,
    729,
    730,
    731,
    732,
    733,
    734,
    735,
    736,
    737,
    738,
    739,
    740,
    741,
    742,
    743,
    744,
    745,
    746,
    747,
    748,
    749,
    750,
    751,
    752,
    753,
    754,
    755,
    756,
    757,
    758,
    759,
    760,
    761,
    762,
    763,
    764,
    765,
    766,
    767,
    768,
    769,
    770,
    771,
    772,
    773,
    774,
    775,
    776,
    777,
    778,
    779,
    780,
    781,
    782,
    783,
    784,
    785,
    786,
    787,
    788,
    789,
    790,
    791,
    792,
    793,
    794,
    795,
    796,
    797,
    798,
    799,
    800,
    801,
    802,
    803,
    804,
    805,
    806,
    807,
    808,
    809,
    810,
    811,
    812,
    813,
    814,
    815,
    816,
    817,
    818,
    819,
    820,
    821,
    822,
    823,
    824,
    825,
    826,
    827,
    828,
    829,
    830,
    831,
    832,
    833,
    834,
    835,
    836,
    837,
    838,
    839,
    840,
    841,
    842,
    843,
    844,
    845,
    846,
    847,
    848,
    849,
    850,
    851,
    852,
    853,
    854,
    855,
    856,
    857,
    858,
    859,
    860,
    861,
    862,
    863,
    864,
    865,
    866,
    867,
    868,
    869,
    870,
    871,
    872,
    873,
    874,
    875,
    876,
    877,
    878,
    879,
    880,
    881,
    882,
    883,
    884,
    885,
    886,
    887,
    888,
    889,
    890,
    891,
    892,
    893,
    894,
    895,
    896,
    897,
    898,
    899,
    900,
    901,
    902,
    903,
    904,
    905,
    906,
    907,
    908,
    909,
    910,
    911,
    912,
    913,
    914,
    915,
    916,
    917,
    918,
    919,
    920,
    921,
    922,
    923,
    924,
    925,
    926,
    927,
    928,
    929,
    930,
    931,
    932,
    933,
    934,
    935,
    936,
    937,
    938,
    939,
    940,
    941,
    942,
    943,
    944,
    945,
    946,
    947,
    948,
    949,
    950,
    951,
    952,
    953,
    954,
    955,
    956,
    957,
    958,
    959,
    960,
    961,
    962,
    963,
    964,
    965,
    966,
    967,
    968,
    969,
    970,
    971,
    972,
    973,
    974,
    975,
    976,
    977,
    978,
    979,
    980,
    981,
    982,
    983,
    984,
    985,
    986,
    987,
    988,
    989,
    990,
    991,
    992,
    993,
    994,
    995,
    996,
    997,
    998,
    999,
    1000,
    1001,
    1002,
    1003,
    1004,
    1005,
    1006,
    1007,
    1008,
    1009,
    1010,
    1011,
    1012,
    1013,
    1014,
    1015,
    1016,
    1017,
    1018,
    1019,
    1020,
    1021,
    1022,
    1023,
  ],
  "string.names": Array [],
  "string.values": Array [],
}
`;
